vpr/src/place/place_log_util.cpp

```diff
@@ -0,0 +1,324 @@
+            total_swap_attempts);
+    VTR_LOG("\tSwaps accepted: %*d (%4.1f %%)\n", num_swap_print_digits,
+            swap_stats.num_swap_accepted, 100 * accept_rate);
+    VTR_LOG("\tSwaps rejected: %*d (%4.1f %%)\n", num_swap_print_digits,
+            swap_stats.num_swap_rejected, 100 * reject_rate);
+    VTR_LOG("\tSwaps aborted: %*d (%4.1f %%)\n", num_swap_print_digits,
+            swap_stats.num_swap_aborted, 100 * abort_rate);
+}
+void PlacementLogPrinter::print_initial_placement_stats() const {
+    if (quiet_) {
+        return;
+    }
+
+    const t_placer_costs& costs = placer_.costs();
+    const t_noc_opts& noc_opts = placer_.noc_opts();
+    const t_placer_opts& placer_opts = placer_.placer_opts();
+    const tatum::TimingPathInfo& critical_path = placer_.critical_path();
+    const std::optional<NocCostHandler>& noc_cost_handler = placer_.noc_cost_handler();
+    std::shared_ptr<const SetupTimingInfo> timing_info = placer_.timing_info();
+    const PlacerState& placer_state = placer_.placer_state();
+
+    VTR_LOG("Initial placement cost: %g bb_cost: %g td_cost: %g\n",
+        costs.cost, costs.bb_cost, costs.timing_cost);
+
+    if (noc_opts.noc) {
+        VTR_ASSERT(noc_cost_handler.has_value());
+        noc_cost_handler->print_noc_costs("Initial NoC Placement Costs", costs, noc_opts);
+    }
+
+    if (placer_opts.place_algorithm.is_timing_driven()) {
+        VTR_LOG("Initial placement estimated Critical Path Delay (CPD): %g ns\n",
+                1e9 * critical_path.delay());
+        VTR_LOG("Initial placement estimated setup Total Negative Slack (sTNS): %g ns\n",
+                1e9 * timing_info->setup_total_negative_slack());
+        VTR_LOG("Initial placement estimated setup Worst Negative Slack (sWNS): %g ns\n",
+                1e9 * timing_info->setup_worst_negative_slack());
+        VTR_LOG("\n");
+        VTR_LOG("Initial placement estimated setup slack histogram:\n");
+        print_histogram(create_setup_slack_histogram(*timing_info->setup_analyzer()));
+    }
+
+    const BlkLocRegistry& blk_loc_registry = placer_state.blk_loc_registry();
+    size_t num_macro_members = 0;
+    for (const t_pl_macro& macro : blk_loc_registry.place_macros().macros()) {
+        num_macro_members += macro.members.size();
+    }
+    VTR_LOG("Placement contains %zu placement macros involving %zu blocks (average macro size %f)\n",
+            blk_loc_registry.place_macros().macros().size(), num_macro_members,
+            float(num_macro_members) / blk_loc_registry.place_macros().macros().size());
+    VTR_LOG("\n");
+
+    sprintf(msg_.data(),
+            "Initial Placement.  Cost: %g  BB Cost: %g  TD Cost %g \t Channel Factor: %d",
+            costs.cost, costs.bb_cost, costs.timing_cost, placer_opts.place_chan_width);
+
+    // Draw the initial placement
+    update_screen(ScreenUpdatePriority::MAJOR, msg_.data(), PLACEMENT, timing_info);
```

Comment:
Why is drawing code in the logging utility? Was this intentional? Or is this just printing the initial placement stats to the screen? If it is the latter, I recommend updating the comment above this line.

Reply:
I'm OK with leaving it here (@soheilshahrouz tells me it does keep the code shorter), but I agree the class should have a comment saying it is controlling drawing and logging.